robot/client/client.go

```diff
@@ -1078,3 +1083,54 @@ func (rc *RobotClient) Version(ctx context.Context) (robot.VersionResponse, erro
 
 	return mVersion, nil
 }
+
+func addVersionMetadataToContext(ctx context.Context) context.Context {
+	info, _ := debug.ReadBuildInfo()
+	deps := make(map[string]*debug.Module, len(info.Deps))
+	for _, dep := range info.Deps {
+		deps[dep.Path] = dep
+	}
+	var apiVersion string
+	if dep, ok := deps["go.viam.com/api"]; ok {
+		apiVersion = dep.Version
+	}
+	version := config.Version
+	if version == "" && config.GitRevision != "" {
+		version = "git-" + config.GitRevision
+	}
+	if version == "" {
+		if dep, ok := deps["go.viam.com/rdk"]; ok {
+			version = dep.Version
+		}
+	}
```

Comment:
I don't really love these alternatives. I think they're pretty verbose and hard to read, and I think there's a significant possibility in cases where `config.Version` doesn't exist that this is someone's local build and so the git commit will be something meaningless to us. I think I would prefer falling back to `unknown build` or something like that. 

If folks disagree I won't necessarily fight for it, but at a minimum we should add a final fallback because as it stands it seems possible to have version never be set and have version metadata be `go;;v0.1.330`

Reply:
I'll remove `config.GitRevision` if it isn't helpful for us. @Otterverse you were the one who mentioned that in Go, the standard is to leave an empty string. What do you think about having `unknown`?